osmdroid-android/src/main/java/org/osmdroid/tileprovider/modules/MapTileSqlCacheProvider.java

```diff
@@ -33,7 +33,7 @@ public class MapTileSqlCacheProvider extends MapTileFileStorageProviderBase {
     // ===========================================================
 
     private final AtomicReference<ITileSource> mTileSource = new AtomicReference<ITileSource>();
-    private SqlTileWriter mWriter;
+    private SqlTileWriter mWriter = new SqlTileWriter();
```

Comment:
I don't follow you here.

Reply:
The default implementation is `SqlTileWriter` and with the use of the new constructor, you can override it with your custom implementation.